src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -204,9 +207,22 @@ def _fetch_client_secret(self, sp_list: list[StoragePermissionMapping]) -> list[
 
             if secret_value:
                 sp_info_with_client_secret[sp_info.application_id] = secret_value
+            elif sp_info.storage_account is not None:
+                if storage_acc_set and sp_info.storage_account in storage_acc_set:
+                    logger.info(
+                        f"Cannot fetch the service principal client_secret for {sp_info.application_id}. "
+                        f"However, this storage account {sp_info.storage_account} is already configured using "
+                        f"access connector."
+                    )
+                else:
+                    logger.info(
+                        f"Cannot fetch the service principal client_secret for {sp_info.application_id}. "
+                        f"An access connector will be used for this storage account {sp_info.storage_account}."
+                    )
+                    storage_acc_no_secret.add(sp_info.storage_account)
             else:
                 logger.info(
-                    f"Cannot fetch the service principal client_secret for {sp_info.application_id}. "
+                    f"Storage account information missing for service principal {sp_info.application_id}. "
                     f"This service principal will be skipped for migration"
```

Comment:
explain how to resolve this:
Rerun migration,
i) Confirm to create access connector instead
ii) After creating SP secret

Reply:
@JCZuurmond  This 'else' block would only be reached if a service principal (SP) doesn't have any storage account (SA) associated with it. Which I am thinking will never happen. If it does, then the SP is of no use, as there is no SA that needs to be connected with a storage credential in this case. I can change the message saying that "skipping this SP as there is no SA associated to it".

The logic is like this -- 
1. First the user will be prompted if they would like create an access connector with a managed identity for each storage account (line 325, existing functionality).
2. If they say yes, then all access connectors will be created (line 297, existing functionality) and a set will be created to keep a track of the SAs which are already configured with access connector (line 312, change in PR).
3. Next it will check the list of SPs in azure_storage_account_info.csv and azure_service_principals table and check if a secret is found (line 206, existing functionality). If not, it will get the SA associated with the SP, check that the access connector is not already added, and then list all new SAs for which access connector needs to be created (line 222, change in PR).
4. It will then prompt the user to accept creating of storage credentials from the SP and access connectors (line 354, change in PR).
5. Create the storage credentials (line 345, existing, and line 358, new).

I can revery back all of my changes if you think it is not needed as we already have a functionality to create access connector for all storage accounts (line 325).

Reply:
> This 'else' block would only be reached if a service principal (SP) doesn't have any storage account (SA) associated with it. Which I am thinking will never happen. If it does, then the SP is of no use, as there is no SA that needs to be connected with a storage credential in this case. I can change the message saying that "skipping this SP as there is no SA associated to it".

Agreed. If I read the code correctly, the following would be reached if the SP has a storage account associated with it, but its credentials are not found:

``` python
                    logger.info(
                        f"Cannot fetch the service principal client_secret for {sp_info.application_id}. "
                        f"An access connector will be used for this storage account {sp_info.storage_account}."
                    )
```

That message can be updated to tell the user to create access connectors instead OR create the SP credentials for ucx to find

> I can revery back all of my changes if you think it is not needed as we already have a functionality to create access connector for all storage accounts (line 325).

yes, to keep the code flow more unidirectional, I prefer to separate the access connector creation from the service principal migration (as much as possible)

Reply:
Thanks, closing this PR as it is no longer needed.